src/main/scala/com/cloudera/sparkts/DateTimeIndex.scala

```diff
@@ -423,6 +478,54 @@ class HybridDateTimeIndex(
     "hybrid," + dateTimeZone.toString + "," +
       indices.map(_.toString).mkString(";")
   }
+
+  override def millisIterator(): Iterator[Long] = {
+    new Iterator[Long] {
+      val indicesIter = indices.iterator
+      var milIter = if (indicesIter.hasNext) indicesIter.next.millisIterator else null
+
+      override def hasNext: Boolean = {
+        if (milIter != null) {
+          if (milIter.hasNext) {
+            true
+          } else if(indicesIter.hasNext) {
+            milIter = indicesIter.next.millisIterator
```

Comment:
In this case consecutive ```hasNext``` calls (before ```next```) will not affect the behavior of the iterator